lib/rspec/mocks/object_reference.rb

```diff
@@ -42,44 +42,67 @@ def self.anonymous_module?(mod)
     # Used when an object is passed to `object_double`.
     # Represents a reference to that object.
     #
-    # @private
+    # @param object
     class DirectObjectReference
       def initialize(object)
         @object = object
       end
 
+      # The objects desription (`#inspect`).
+      # @return String
       def description
         @object.inspect
       end
 
+      # Not relevant to a direct object reference.
       def const_to_replace
         raise ArgumentError,
               "Can not perform constant replacement with an object."
       end
 
+      # The target of the verifying double (the object's class).
+      #
+      # @return Class
       def target
         @object.class
       end
 
+      # Always returns true for an object as the class is defined.
+      #
+      # @return true
       def defined?
         true
       end
 
+      # Performs the callback straight away as the object is already
+      # loaded.
```

Comment:
This doc (and the others here) describes the specific implementation of this method rather than explaining its use as part of the generic reference interface (of which there are multiple implementations).  I think it might be better to document it with something like:

``` ruby
# Yields if the reference target is loaded, providing a generic mechanism
# to optionally run a bit of code only when a reference's target is loaded.
# This specific implementation always yields because direct references
# are always loaded.
```

Thoughts?  Could be good to provide a similar treatment for the other methods, so that users who read the docs get some insight into the purpose of these methods.  Otherwise, `when_loaded` seems like a pretty pointless method...